python/plugins/processing/algs/gdal/extractprojection.py

```diff
@@ -100,7 +100,11 @@ def processAlgorithm(self, parameters, context, feedback):
         raster = None
         rasterDS = None
 
-        outFileName = os.path.splitext(str(rasterPath))[0]
+        inFileName = os.path.splitext(str(rasterPath))
+        outFileName = inFileName[0]
+        # this is not a good idea as it won't work with an extension like .jpeg
+        # outFileExt = '.' + inFileName[1][1:4:2] + 'w'
```

Comment:
Do you have a list of possible extension?
This hack should be only available for file type using geo referencing (and .wld for the other one)

Reply:
I don't know what you mean by "file type using geo referencing".
The traditional form works for any file type - .bpw for BMP, .tfw for TIFF, .jgw for JPEG, .pgw for PNG, .gfw for GIF, etc.  I don't believe there is a situation where .wld is valid but the traditional form is not.

Reply:
I guess the issue would be for a file with no extension at all (code doesn't work), or an extension that is too short.  Is that what you're getting at?
Are you saying we should check the extension against a list of all known file extensions for raster images?
I think it would make more sense just to use .wld if the extension is less than three characters (plus the .)

Reply:
> I think it would make more sense just to use .wld if the extension is less than three characters (plus the .)

Actually, the traditional form still makes sense for a two character (plus .) extension, although I'm not sure if anyone will ever encounter one.

Reply:
So, it's sounds good to me.